### PR TITLE
RavenDB-18066 Import/Export indexes

### PR DESCRIPTION
**Pull request by kalczur**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18066/Consider-adding-a-quicker-way-to-copy-single-and-multiple-indexes-between-databases

### Additional description

It adds option to inport/export indexes from/to file or a database.

For import/export the default mode is "database". If there is no other db, then mode is "a file".

Added to common:
- Components:
	- FileDropzone
	- CheckboxSelectAll
- Hooks:
	- useCheckboxes

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [x] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Comment by kalczur** [comment]
@ppekrol 
In the "List of indexes" view we do not have access to definitions (only to stats), so I added an endpoint.

Please, check `IndexHandler.cs` and add a endpoint for sharded db.

I also wanted to ask if we should allow export/import of specific node/shard?

File: src/Raven.Studio/typescript/components/common/EmptySet.tsx [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
+    const { children, icon, color, className, compact, iconSize = "xl" } = props;
     const defaultIcon: IconName = "empty-set";
 
     return (
-        <div className={classNames("empty-set text-center mb-2 mx-auto", className)}>
-            <Icon icon={icon || defaultIcon} color={color} className="icon-xl" margin="m-0" />
-            <div className="lead">{children}</div>
+        <div className={classNames("empty-set", className, compact ? "compact my-2" : "mb-2 mx-auto text-center")}>
+            <Icon icon={icon || defaultIcon} color={color} margin="m-0" size={compact ? "sm" : iconSize} />
+            <div className={classNames("mb-0", compact ? "lh-1" : "lead")}>{children}</div>
```
Comment (kalczur) [inline src/Raven.Studio/typescript/components/common/EmptySet.tsx:25]: It is copy-paste from v6.1

**Comment by kalczur** [comment]
I will check if it works properly with certificates

**Comment by kalczur** [comment]
If the user has Read Only access, the import button is hidden, and during export only databases to which he has Write access are visible.

If the user does not have access to the database, he simply cannot see them in select for export.

**Comment by ayende** [comment]
What happens when I have read/write, but not db admin?
I should be able to create JS indexes, but not C# ones.

**Comment by ppekrol** [comment]
Described to @kalczur in details how it works underneath, this will be handled

**Comment by kalczur** [comment]
@ayende Okay, I'll handle that too

**Review by ml054 — APPROVED** [review]

**Review by ml054 — APPROVED** [review]

**Comment by mateuszbartosik** [comment]
Moved warning icon to the left so it won't get confused with Faulty index
image

**Pull request merged**